src/raw_decompiled/HEROES2W.c

```diff
@@ -29331,8 +29331,7 @@ void __thiscall game::SetupTowns(game *this)
 // 429E30: using guessed type int numSpellsOfLevel[5];
 
 //----- (0042A730) --------------------------------------------------------
-void *__thiscall game::ProcessOnMapHeroes(game *this)
-{
+void* game::ProcessOnMapHeroes() {

```

Comment:
This is a void fuction, not a void*. Change it in the decompiler and it will be simplified.

How can you tell?

Its return value doesn't really make sense. Its called only once, and its return value is ignored.

But, if you really want to be sure, check the assembly. For all the functions that actually return a value, you'll see it deliberately move a value into eax before the function exits. If it doesn't, it's probably a void function.

Reply:
This was later corrected.